tests/math_builtin_api/modules/test_generator.py

```diff
@@ -118,22 +141,51 @@ def generate_function_call(sig, arg_names, arg_src):
     fc = fc[:-1] + ");\n"
     return fc
 
+def generate_function_private_call(sig, arg_names, arg_src, types):
+    fc = arg_src
+    fc += "$RETURN_TYPE res = " + sig.namespace + "::" + sig.name + "("
+    for arg_n in arg_names:
+        fc += arg_n + ","
+    fc = fc[:-1] + ");\n"
+    fc += "return privatePtrCheck<" + sig.ret_type + ","
+    fc += extract_type(types[sig.arg_types[-1]]).name + ">("
+    fc += "res, multiPtrSourceData);\n"
+    return fc
```

Comment:
Since you are already using a lot https://docs.python.org/3/library/string.html#template-strings, why not using it?
Also just `"""` strings with https://docs.python.org/3/library/string.html#format-string-syntax are pretty handy.

More generally https://docs.python.org/3/howto/functional.html is useful.